airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/javabase/IntegrationCliParser.java

```diff
@@ -0,0 +1,191 @@
+          .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+      case READ -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_STATE_KEY).desc(JavaBaseConstants.ARGS_PATH_DESC).hasArg(true).build());
+      }
+      case WRITE -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+
+    final CommandLine parsed = runParse(options, args, command);
+    Preconditions.checkNotNull(parsed);
+    final Map<String, String> argsMap = new HashMap<>();
+    for (final Option option : parsed.getOptions()) {
+      argsMap.put(option.getLongOpt(), option.getValue());
+    }
+
+    switch (command) {
+      case SPEC -> {
+        return IntegrationConfig.spec();
+      }
+      case CHECK -> {
+        return IntegrationConfig.check(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case DISCOVER -> {
+        return IntegrationConfig.discover(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case READ -> {
+        return IntegrationConfig.read(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_STATE_KEY));
+      }
+      case WRITE -> {
+        return IntegrationConfig.write(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY));
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+  }
+
+  private static CommandLine runParse(Options options, String[] args, Command command) {
+    final CommandLineParser parser = new DefaultParser();
+    final HelpFormatter helpFormatter = new HelpFormatter();
+
+    try {
+      return parser.parse(options, args);
+    } catch (ParseException e) {
+      LOGGER.error(e.toString());
+      helpFormatter.printHelp(command.toString().toLowerCase(), options);
+      return null;
+    }
+  }
+
+  // https://stackoverflow.com/questions/33874902/apache-commons-cli-1-3-1-how-to-ignore-unknown-arguments
+  private static class RelaxedParser extends DefaultParser {
```

Comment:
Is there a reason not to be strict when specifying inputs, so we can be sure we're able to orchestrate without unexpected inputs and behavior?

Reply:
yeah. the apache cli parser is a bit restrictive. what i want to do is make sure the correct args are present for the chosen command. the only way i could figure to to that with this parse was be doing it in two passes.
1. the first time is _just_ to get what command is used. using the option group guarantees only one command is specified. that is when we use this relaxed parser, because the goal is explicitly to find the command and then ignore everything else. 
1. the second pass re-parses all of the args and makes sure everything that is required is present. this does _not_ use the relaxed parser and makes sure that every arg passed to the program should be there.